core/services/offchainreporting/contract_tracker_test.go

```diff
@@ -26,6 +31,109 @@ func mustNewContract(t *testing.T, address gethCommon.Address) *offchain_aggrega
 	return contract
 }
 
+func Test_OCRContractTracker_LatestBlockHeight(t *testing.T) {
+	t.Parallel()
+
+	t.Run("on L2 chains, always returns 0", func(t *testing.T) {
+		tracker := offchainreporting.NewOCRContractTracker(
+			mustNewContract(t, cltest.NewAddress()),
+			nil,
+			nil,
+			nil,
+			nil,
+			42,
+			*logger.Default,
+			nil,
+			nil,
+			chains.OptimismMainnet,
+			nil,
+		)
+
+		l, err := tracker.LatestBlockHeight(context.Background())
+		require.NoError(t, err)
+
+		assert.Equal(t, uint64(0), l)
+	})
+
+	t.Run("before first head incoming, looks up on-chain", func(t *testing.T) {
+		ethClient := new(mocks.Client)
+
+		ethClient.On("HeaderByNumber", mock.AnythingOfType("*context.cancelCtx"), (*big.Int)(nil)).Return(&models.Head{Number: 42}, nil)
+
+		tracker := offchainreporting.NewOCRContractTracker(
+			mustNewContract(t, cltest.NewAddress()),
+			nil,
+			nil,
+			ethClient,
+			nil,
+			42,
+			*logger.Default,
+			nil,
+			nil,
+			chains.EthMainnet,
+			nil,
+		)
+
+		l, err := tracker.LatestBlockHeight(context.Background())
+		require.NoError(t, err)
+
+		assert.Equal(t, uint64(42), l)
+	})
+
+	t.Run("Before first head incoming, on client error returns error", func(t *testing.T) {
+		ethClient := new(mocks.Client)
+
+		tracker := offchainreporting.NewOCRContractTracker(
+			mustNewContract(t, cltest.NewAddress()),
+			nil,
+			nil,
+			ethClient,
+			nil,
+			42,
+			*logger.Default,
+			nil,
+			nil,
+			chains.EthMainnet,
+			nil,
+		)
+
+		ethClient.On("HeaderByNumber", mock.AnythingOfType("*context.cancelCtx"), (*big.Int)(nil)).Return(nil, nil).Once()
+
+		_, err := tracker.LatestBlockHeight(context.Background())
+		assert.EqualError(t, err, "got nil head")
+
+		ethClient.On("HeaderByNumber", mock.AnythingOfType("*context.cancelCtx"), (*big.Int)(nil)).Return(nil, errors.New("bar")).Once()
+
+		_, err = tracker.LatestBlockHeight(context.Background())
+		assert.EqualError(t, err, "bar")
+
+		ethClient.AssertExpectations(t)
+	})
+
+	t.Run("after first head incoming, uses cached value", func(t *testing.T) {
+		tracker := offchainreporting.NewOCRContractTracker(
+			mustNewContract(t, cltest.NewAddress()),
+			nil,
+			nil,
+			nil,
+			nil,
+			42,
+			*logger.Default,
+			nil,
+			nil,
+			chains.EthMainnet,
+			nil,
```

Comment:
hmm it's quite a lot of cryptic values, repeated a few times (in existing tests as well) - might be good to create some helper function?